### ISSUE TITLE
`missing_safety_doc` lint doesn't trigger on `extern` block function declarations

### ISSUE THREAD

**Issue by Lokathor** [rust-lang/rust-clippy#9977]
```rust
extern "C" {
  pub fn __aeabi_memcpy(dest: *mut u8, src: *const u8, byte_count: usize);
}
```

This doesn't trigger the `missing_safety_doc` lint, though it probably should.
